manifests/init.pp

```diff
@@ -34,8 +64,16 @@
 #   systemd-networkd take precedence over this setting. requires puppetlabs-inifile
 class systemd (
   Hash[String, Hash[String, Any]]  $service_limits   = {},
-  Boolean                          $manage_resolved  = false,
-  Enum['stopped','running']        $resolved_ensure  = 'running',
+  Boolean                                            $manage_resolved   = false,
+  Enum['stopped','running']                          $resolved_ensure   = 'running',
+  Optional[Variant[Array,String]]                    $dns               = undef,
+  Optional[Variant[Array,String]]                    $fallback_dns      = undef,
+  Optional[Variant[Array,String]]                    $domains           = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]         $llmnr             = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]         $multicast_dns     = undef,
+  Optional[Variant[Boolean,Enum['allow-downgrade']]] $dnssec            = undef,
+  Optional[Boolean]                                  $cache             = undef,
```

Comment:
In my opionion, everything that can be a boolean should default to false or true, so it doesn't need to be Optional.